src/Content/ContactSelector.php

```diff
@@ -119,15 +119,29 @@ public static function networkToName($s, $profile = "")
 	public static function gender($current = "", $suffix = "")
 	{
 		$o = '';
-		$select = ['', L10n::t('Male'), L10n::t('Female'), L10n::t('Currently Male'), L10n::t('Currently Female'), L10n::t('Mostly Male'), L10n::t('Mostly Female'), L10n::t('Transgender'), L10n::t('Intersex'), L10n::t('Transsexual'), L10n::t('Hermaphrodite'), L10n::t('Neuter'), L10n::t('Non-specific'), L10n::t('Other'), L10n::t('Undecided')];
+		$select[''] = '';
```

Comment:
I'm more familiar with the following array declaration syntax:
```php
$select = [
  ''     => '',
  'Male' => L10n::t('Male'),
...
];
```
Additionally. I like the arrows to be aligned based on the longest or second longest key in the declared array.

Reply:
would you do anything to the double `'Male' => L10n::t('Male'),`
To me copying is a catch for future related bugs.

Reply:
What double are you talking about?

Reply:
the key that is copied into `L10n::t();` as parameter.

Reply:
The list isn't moving that often and is small enough to be quickly checked manually. If you want to be that cautious, you can write something like this:
```php
$select = [];
$choices = [
  '',
  'Male',
  ...
];
foreach($choices as $choice) {
  $select[$choice] = L10n::t($choice);
}
```

Reply:
The issue with that is that the translation string extraction won't contain the actual strings which need to be translated, only a variable.

Reply:
For now only the profile-view on friendica has actually been translated. I will tackle the federation protocols soon making them to translate it before sending if necessary.

Reply:
@redmatrix Do you mean that at the next pass of gettext, the existing translations would be wiped out because they wouldn't be found? In this case, we would have to keep the syntax in https://github.com/friendica/friendica/pull/5751#discussion_r216170614

Reply:
All I can say for sure is that it is difficult to translate a variable. That's why for the most part translations are performed on "constant" strings. If you're satisfied that t($something) will be translated correctly for any possible value of $something you can ignore this warning. The code has changed dramatically since I last worked on it, but  the  ability to translate variables  is sort of a binary decision. Either you provide a mechanism to translate every possible value of the variable, or you need to enumerate each of them for the translation service.

Reply:
Got it, thanks for the elaboration.